### PR TITLE
Iss32 feedback model api

### PR DESCRIPTION
**Pull request by Nandu-D**
Interview model - Added feedback
Interview controller - Added routes for creating and retrieving feedback
Interview Service - Logic for creating and retrieving feedback

### CODE REVIEW

File: server/interview/interview.model.js [hunk 1]
```diff
@@ -11,6 +11,82 @@ let InterviewSchema = new Schema({
...
+          //What are some things this candidate can improve on (the more specific the better)
+          type: String,
+        },
+        recommendations: {
+          //Any recommendations on resources that can help this candidate improve?
+          type: String,
+        },
+        anythingElse: {
+          type: String,
+        },
```
Comment (skassam21) [inline server/interview/interview.model.js]: FeedbackSchema can be separated

File: server/interview/interview.service.js [hunk 2]
```diff
@@ -184,6 +193,94 @@ async function retrieveRelevantInterviewInformation(userId, interviewIds) {
   return relevantDetails;
 }
 
+/**
+ * Creates feedback from this user to the other participant
+ */
+async function createFeedback(userId, interviewId, feedbackBody) {
+  let feedbackReceived = {};
```
Comment (skassam21) [inline server/interview/interview.service.js]: This should be const

File: server/interview/interview.service.js [hunk 3]
```diff
@@ -184,6 +193,94 @@ async function retrieveRelevantInterviewInformation(userId, interviewIds) {
...
+    const problemSolvingSkills = feedbackBody.review.problemSolvingSkills;
+
+    if (communicationSkills) review.communicationSkills = communicationSkills;
+    if (codeEfficiency) review.codeEfficiency = codeEfficiency;
+    if (speed) review.speed = speed;
+    if (debuggingSkills) review.debuggingSkills = debuggingSkills;
+    if (problemSolvingSkills)
+      review.problemSolvingSkills = problemSolvingSkills;
+
+    feedbackReceived.review = review;
```
Comment (skassam21) [inline server/interview/interview.service.js]: feedbackReviewed.review = feedbackBody.review; OR:

```
for (const key in feedbackBody.review) {
    if (feedbackBody.review[key]) review[key] = feedbackBody.review[key];
}
```

File: server/interview/interview.model.js (hunk 1 above)
Comment (Nandu-D) [reply server/interview/interview.model.js]: Schema is separated into a review schema

File: server/interview/interview.service.js (hunk 2 above)
Comment (Nandu-D) [reply server/interview/interview.service.js]: I rewrote it and now the variable gets assigned a second time. So I had to use 'let' again to allow for reassignment

File: server/interview/interview.service.js (hunk 3 above)
Comment (Nandu-D) [reply server/interview/interview.service.js]: Now the review as a whole is retrieved and sent

**Review by skassam21 — APPROVED** [review]
Much cleaner!

**Review by abohakobyan — APPROVED** [review]
Tested, Works~!

**Pull request merged**